File: tools/send_marketing_news.py
#!/usr/bin/env python3
"""
Envia o email diario de noticias de marketing e IA da LocalRise via Resend API.
Uso: python3 tools/send_marketing_news.py /tmp/subject.txt /tmp/body.html
"""
import sys
import json
import urllib.request
import urllib.error

RESEND_API_KEY = "re_4BhjbvVa_NgYbnREDjNmaiWwtpXPCbt2o"
FROM_EMAIL = "LocalRise <noticias@noticias.localriseadvisory.com>"
RECIPIENTS = ["digui.slater@gmail.com", "julieta.slater@gmail.com"]


def send_email(subject, body_html):
    payload = json.dumps({
        "from": FROM_EMAIL,
        "to": RECIPIENTS,
        "subject": subject,
        "html": body_html,
    }).encode("utf-8")

    req = urllib.request.Request(
        "https://api.resend.com/emails",
        data=payload,
        headers={
            "Authorization": f"Bearer {RESEND_API_KEY}",
            "Content-Type": "application/json",
        },
        method="POST",
    )

    try:
        with urllib.request.urlopen(req) as resp:
            result = json.loads(resp.read().decode())
            print("Email enviado com sucesso! ID: " + result.get("id", "?"))
            print("Destinatarios: " + ", ".join(RECIPIENTS))
    except urllib.error.HTTPError as e:
        error_body = e.read().decode()
        print(f"Erro ao enviar email: HTTP {e.code}")
        print(error_body)
        sys.exit(1)


if __name__ == "__main__":
    if len(sys.argv) != 3:
        print("Uso: python3 send_marketing_news.py <subject_file> <body_html_file>")
        sys.exit(1)

    with open(sys.argv[1], encoding="utf-8") as f:
        subject = f.read().strip()

    with open(sys.argv[2], encoding="utf-8") as f:
        body_html = f.read()

    send_email(subject, body_html)

File: tools/send_marketing_news.js
#!/usr/bin/env node
/**
 * Envia email de noticias via Resend API.
 * Uso: node tools/send_marketing_news.js <subject_file> <body_html_file>
 */
const https = require("https");
const fs = require("fs");

const RESEND_API_KEY = "re_4BhjbvVa_NgYbnREDjNmaiWwtpXPCbt2o";
const FROM = "LocalRise <noticias@noticias.localriseadvisory.com>";
const TO = ["digui.slater@gmail.com", "julieta.slater@gmail.com", "contato@localriseadvisory.com"];

if (process.argv.length !== 4) {
  console.error("Uso: node send_marketing_news.js <subject_file> <body_html_file>");
  process.exit(1);
}

const subject = fs.readFileSync(process.argv[2], "utf8").trim();
const html = fs.readFileSync(process.argv[3], "utf8");

const payload = JSON.stringify({ from: FROM, to: TO, subject, html });

const options = {
  hostname: "api.resend.com",
  path: "/emails",
  method: "POST",
  headers: {
    Authorization: `Bearer ${RESEND_API_KEY}`,
    "Content-Type": "application/json",
    "Content-Length": Buffer.byteLength(payload),
  },
};

const req = https.request(options, (res) => {
  let body = "";
  res.on("data", (chunk) => (body += chunk));
  res.on("end", () => {
    if (res.statusCode === 200 || res.statusCode === 201) {
      console.log("Email enviado com sucesso!");
      console.log("Para:", TO.join(", "));
      console.log("ID:", JSON.parse(body).id);
    } else {
      console.error("Erro ao enviar. Status:", res.statusCode);
      console.error("Resposta:", body);
      process.exit(1);
    }
  });
});

req.on("error", (e) => {
  console.error("Erro de conexao:", e.message);
  process.exit(1);
});

req.write(payload);
req.end();
